### PR TITLE
Create InvoiceItemRepository

### LINKED ISSUE #25: Iteration 3.1 Invoice Item Repository
InvoiceItemRepository
Invoice items are how invoices are connected to items. A single invoice item connects a single item with a single invoice.

The InvoiceItemRepository is responsible for holding and searching our InvoiceItem instances. It offers the following methods:

all - returns an array of all known InvoiceItem instances
find_by_id - returns either nil or an instance of InvoiceItem with a matching ID
find_all_by_item_id - returns either [] or one or more matches which have a matching item ID
find_all_by_invoice_id - returns either [] or one or more matches which have a matching invoice ID
create(attributes) - create a new InvoiceItem instance with the provided attributes. The new InvoiceItem’s id should be the current highest InvoiceItem id plus 1.
update(id, attribute) - update the InvoiceItem instance with the corresponding id with the provided attributes. Only the invoice_item’s quantity and unit_price can be updated. This method will also change the invoice_item’s updated_at attribute to the current time.
delete(id) - delete the InvoiceItem instance with the corresponding id
The data can be found in data/invoice_items.csv so the instance is created and used like this:

```rb
sales_engine = SalesEngine.from_csv(:invoice_items => "./data/invoice_items.csv")
invoice_item = sales_engine.invoice_items.find_by_id(6)
# => <InvoiceItem>
InvoiceItem
``` 
3 points

### PR DESCRIPTION
**Pull request by NoahZinter**
### Changes

- Description of changes
Created Invoice Item Repository class and spec. All tests passing, complies with spec harness. 

### Requested review on...
- Specifics on, logic, naming schemes, etc.
Should line up very tightly with other Repository classes. 
### Checklist
- [x] Tests pass
- [x]  Ready to merge

closes #25

### CODE REVIEW

**Review by redferret — APPROVED** [review]
This all looks great!

File: lib/invoice_item_repository.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @invoice_items.select do |invoice_item|
+      invoice_item.invoice_id == id
+    end
+  end
+
+  def create_first_invoice_item(attributes)
+    new_invoice_item = InvoiceItem.new(attributes)
+    new_invoice_item.update_id(1)
+    @invoice_items << new_invoice_item
+  end
```
Comment (mkrumholz) [inline lib/invoice_item_repository.rb:38]: Looks good! I think we're a bit inconsistent with this across classes -- not a problem here, but maybe when we deal with inheritance we can ensure all the repo classes are in line.

Comment (mkrumholz) [reply lib/invoice_item_repository.rb:38]: (Don't think we have this guard in Merchant class and I'm not totally sure we need it anywhere)

File: spec/mock_data_spec.rb [hunk 2]
```diff
@@ -8,10 +8,10 @@
...
         name: 'Example 1'
       },
-      {
-        id: 11,
-        name: 'Example 2'
-      }]
+                {
+                  id: 11,
+                  name: 'Example 2'
+                }]
```
Comment (mkrumholz) [inline spec/mock_data_spec.rb]: ⛏ What's going on here? Looks off to me. Seems like just some odd formatting.

File: lib/invoice_item.rb [hunk 3]
```diff
@@ -20,4 +20,20 @@ def initialize(details)
...
+  end
+
+  def update_quantity(quantity)
+    @quantity = quantity
+  end
+
+  def update_unit_price(unit_price)
+    @unit_price = unit_price
+  end
+
```
Comment (mkrumholz) [inline lib/invoice_item.rb]: You're going to need `unless` guards on these to pass the spec harness

**Review by mkrumholz — APPROVED** [review]
Made some nit-picky comments and noted one spot where you're going to need some updates for the spec harness, I believe. But should be all good beyond those.

File: lib/invoice_item.rb (hunk 3 above)
Comment (NoahZinter) [reply lib/invoice_item.rb]: I have return nil guards in the item_repo class in order to pass spec harness for the nil inputs. Would it be better to have them down in the invoice_item class?

File: spec/mock_data_spec.rb (hunk 2 above)
Comment (NoahZinter) [reply spec/mock_data_spec.rb]: huh, not sure, this must be a rubocop thing? I didn't manually go to change anything in mock_data_spec.

**Pull request merged**